presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -105,10 +108,26 @@ public ConnectorPartitionResult getPartitions(ConnectorSession session, Connecto
                         log.warn("No leader for partition %s/%s found!", metadata.topic(), part.partitionId());
                     }
                     else {
+                        ImmutableList.Builder<HostAddress> partitionNodesBuilder = ImmutableList.builder();
+                        HostAddress partitionLeader = brokerToHostAddress(leader);
+
+                        if (connectorConfig.isForceTalkToLeader()) {
+                            // make sure the first broker is the partition leader
+                            partitionNodesBuilder.add(partitionLeader);
+                            for (Broker broker : part.isr()) {
```

Comment:
Should we really be adding the rest of the brokers? I don't have a good understanding of this code, but it seems like a flag named "forceTalkToLeader" shouldn't include anything that's not a leader